staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -204,6 +213,10 @@ func (f *fancyResponseWriterDelegator) Flush() {
 func (f *fancyResponseWriterDelegator) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	// fake a response status before protocol switch happens
 	f.processCode(http.StatusSwitchingProtocols)
+
+	// TODO(caoshufeng): we can't call WriteHeader before Hijack(), because WriteHeader may be called later:
```

Comment:
> Agree, a header should be set and then a comment should be added that it's ignored in certain cases
Also, I think it's worth mentioning in the release notes that Audit-Id http header is not set for some requests (and describe in short for which)

+1